numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -38,6 +38,26 @@
 #include "npy_cblas.h"
 #include "npy_buffer.h"
 
+
+/*
+ * Define a stack allocated dummy array with only the minimum information set:
+ *   1. The descr, the main field interesting here.
+ *   2. The flags, which are needed for alignment;.
+ *   3. The type is set to NULL and the base is the original array, if this
+ *      is used within a subarray getitem to create a new view, the base
+ *      must be walked until the type is not NULL.
```

Comment:
OK, this got more complex, but for now seemed like the easiest. The third point is only necessary for `VOID_getitem` recursion, so it would be plausible to create two versions for creating the stack array...

Reply:
I initially thought that the base iteration is handled implicitly in the normal base setting code, but, there are always cases that do not work (when subclasses are involved).

This is probably better anyway, keep the strangeness confined to a single place...